evm/hyperliquid.js

```diff
@@ -0,0 +1,225 @@
+
+    const useBig = blockSize === 'big';
+    const network = chain.networkType;
+
+    printInfo('Block size', blockSize);
+    printInfo('Network', network);
+
+    const action = { type: 'evmUserModify', usingBigBlocks: useBig };
+    const nonce = Date.now();
+    const signature = await signL1Action(wallet, action, null, nonce, chain);
+    const payload = { action, signature, nonce };
+    const endpoint = `${chain.hypercore.url}/exchange`;
+    const result = await httpPost(endpoint, payload);
+
+    if (result.status !== 'ok') {
+        throw new Error(result);
+    }
+
+    printInfo('Result', result);
+    return result;
+}
+
+async function deployer(wallet, chain, options) {
+    const [tokenId] = options.args;
+    validateParameters({
+        isNonEmptyString: { tokenId },
+    });
+
+    const contracts = chain.contracts;
+    const interchainTokenFactoryAddress = contracts.InterchainTokenFactory?.address;
+    const interchainTokenServiceAddress = contracts.InterchainTokenService?.address;
+
+    validateParameters({
+        isValidAddress: { interchainTokenFactoryAddress, interchainTokenServiceAddress },
+    });
+
+    const IInterchainTokenService = getContractJSON('IInterchainTokenService');
+
+    const interchainTokenService = new Contract(interchainTokenServiceAddress, IInterchainTokenService.abi, wallet);
+
+    const tokenAddress = await interchainTokenService.registeredTokenAddress(tokenId);
+    printInfo('Token address', tokenAddress);
+
+    try {
+        const TokenContract = getContractJSON('HyperliquidInterchainToken');
+        const token = new Contract(tokenAddress, TokenContract.abi, wallet);
+
+        const currentDeployer = await token.deployer();
+        printInfo('Current deployer', currentDeployer);
+    } catch (error) {
+        throw error;
+    }
+}
+
+async function updateTokenDeployer(wallet, chain, options) {
+    const [tokenId, deployer] = options.args;
+    validateParameters({
+        isNonEmptyString: { tokenId },
+        isValidAddress: { deployer },
+    });
+
+    const contracts = chain.contracts;
+    const interchainTokenFactoryAddress = contracts.InterchainTokenFactory?.address;
+    const interchainTokenServiceAddress = contracts.InterchainTokenService?.address;
+
+    validateParameters({
+        isValidAddress: { interchainTokenFactoryAddress, interchainTokenServiceAddress },
+    });
+
+    const IInterchainTokenService = getContractJSON('IInterchainTokenService');
+    const interchainTokenService = new Contract(interchainTokenServiceAddress, IInterchainTokenService.abi, wallet);
+
+    const tokenAddress = await interchainTokenService.registeredTokenAddress(tokenId);
+    printInfo('Token address', tokenAddress);
+
+    const ServiceContract = getContractJSON('HyperliquidInterchainTokenService');
+    const service = new Contract(interchainTokenServiceAddress, ServiceContract.abi, wallet);
+
+    const TokenContract = getContractJSON('HyperliquidInterchainToken');
```

Comment:
```suggestion
    const HyperliquidInterchainToken = getContractJSON('HyperliquidInterchainToken');
```
the pattern is to use the contract name as-is